sdk/keyvault/azure-security-keyvault-keys/src/delete_key_operation.cpp

```diff
@@ -19,8 +20,14 @@ Azure::Security::KeyVault::Keys::DeleteKeyOperation::PollInternal(
   std::unique_ptr<Azure::Core::Http::RawResponse> rawResponse;
   if (!IsDone())
   {
-    rawResponse = m_pipeline->Send(
-        context, Azure::Core::Http::HttpMethod::Get, {_detail::DeletedKeysPath, m_value.Name()});
+    try
+    {
+      rawResponse = m_keyClient->GetDeletedKey(m_value.Name(), context).RawResponse;
+    }
+    catch (Azure::Core::RequestFailedException& error)
+    {
+      rawResponse = std::move(error.RawResponse);
```

Comment:
Why are we wrapping this in a try-catch and not bubbling the exception to the user?

Reply:
Exception could mean that the operation is not yet completed.  Below code would check that and re-throw if the exception is not one of the ones used for operation-non completed